pywbem_mock/_namespaceprovider.py

```diff
@@ -0,0 +1,290 @@
+#
+# (C) Copyright 2020 InovaDevelopment.com
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+This module implements a user-defined provider for the class CIM_Namespace
+
+This  provider that allows users
+access to the list of namespaces and to be able to control existence of
+namespaces through create and delete instances of the class CIM_Namespace.
+"""
+
+from copy import deepcopy
+
+from pywbem import CIMError, CIMInstance, CIM_ERR_NOT_SUPPORTED, \
+    CIM_ERR_INVALID_PARAMETER
+
+from pywbem._nocasedict import NocaseDict
+
+from pywbem._utils import _format
+
+# TODO: Should be able to import from pywbem_mock
+from ._defaultinstanceprovider import InstanceWriteProvider
+
+from .config import OBJECTMANAGERNAME, SYSTEMNAME, SYSTEMCREATIONCLASSNAME
+
+
+def create_cimnamespace_instance(conn, namespace, interop_namespace, klass):
+    """
+    Build and execute CreateInstance for an instance of CIM_Namespace using
+    the `namespace` as the value of the name property in the instance. The
+    instance is created in the `interop_namespace`
+
+    Parameters:
+
+      namespace
+        Namespace that this instance defines
+
+      interop_namespace
+        Interop namespace for this environment.
+
+      klass
+
+    Raises:
+
+        CIMERROR: For errors encountered with CreateInstance
+
+    """
+    properties = NocaseDict(
+        [('Name', namespace),
+         ('CreationClassName', klass.classname),
+         ('ObjectManagerName', OBJECTMANAGERNAME),
+         ('ObjectManagerCreationClassName', 'CIM_ObjectManager'),
+         ('SystemName', SYSTEMNAME),
+         ('SystemCreationClassName', SYSTEMCREATIONCLASSNAME)])
+
+    new_instance = CIMInstance.from_class(klass, namespace, properties)
```

Comment:
Passing the `namespace` parameter to `from_class()` is not needed, because it will only be used if `include_path` is set to True. It is not incorrect to pass it, but misleading to the reader. Change the call to:
```
from_class(klass, property_values=properties)
```

Reply:
And I wrote that method.  I have no memory left. DONE